### ISSUE TITLE
feat-req: Expose superform stores on new method

### ISSUE THREAD

**Issue by saturnonearth** [huntabyte/formsnap#43]
It would be nice if we could have access to the stores that are created via superforms, so we bind that data or do whatever we wish with it.

I have a global derived svelte store that keeps track of all forms or actions being currently submitted, so I usually bind the `$submitting` superform store to my own custom store, which then get's picked up by the derived store.

As of right now, we only have access to these values from within the `Form.Root`, so the only options are either to...

a) Create a new component and insert it as a child
or
b) set the store value within the `onSubmit` superform event handler

You said...

> The big thing is that basically all the options, etc. are all applied beforehand so it can't be reinstantiated

So maybe there could be a method that could be called on the `Form.Root` so when the `Form` is initialized, we could have access to any of these stores and bind them if we wanted.

Something like...
 
```js
<Form.Root onInit={ ({ submitting, delayed, tainted, form }) => {
    // Set what you want or bind to or own stores...
    ...
}}
</Form.Root>
```

What do you think?

**Comment by huntabyte** [comment]
Hey @saturnonearth, if you could expand on your example code block there for how you'd imagine it working, it'll give me a better idea of the ask/specifics behind it!

**Comment by saturnonearth** [comment]
> Hey @saturnonearth, if you could expand on your example code block there for how you'd imagine it working, it'll give me a better idea of the ask/specifics behind it!

```js
<script>
const formSubmitting = writable(false);
const form: formSchema = writable({});

// Compare the data from what the customer entered, so we can keep track of whether data is old or stale
$: $headingFormStale = !compare(itemSaved, $form);
</script>

<Form.Root onInit={ ({ submitting, delayed, tainted, form }) => {
     $formSubmitting = submitting;
     $form = form;
  }

  ...
  </Form.Root>
  ```
  
Upon writing this out, I realize it's a bit weird. I guess my end goal is just to have access to the Superform stores from outside the `Form.Root`, as the stores give a lot of "super" useful info that is often useful for keeping track of state outside of the form itself.

**Comment by huntabyte** [comment]
Okay, I see what you're getting at. I know this is possible and how to do it, I'm just trying to think of the implications of doing so. They wouldn't be stores, but they would update when the stores update, you'd have to do something like:

```svelte
<script>
	let formSubmitting = false;
</script>

<Form.Root bind:formSubmitting />
```

which is similar to how things are handled in `bits-ui` with how we update the prop's value if the store changes.

**Comment by huntabyte** [comment]
Going to be revisiting this in relation to #69 as I think the two are more or less related, at least from first glance. I have some ideas that I'll put up in a PR once I have some time for you all to check out!

**Comment by huntabyte** [comment]
Closed by #83 :)
